build.gradle.kts

```diff
@@ -29,11 +32,26 @@ java {
     withSourcesJar()
 }
 
-tasks.withType(Javadoc::class).configureEach {
-    val customArgs = projectDir.resolve("javadoc-silence.txt")
-    customArgs.writeText("""-Xdoclint:none
-    """.trimIndent())
-    options.optionFiles?.add(customArgs)
+tasks {
+    withType<KotlinCompile> {
+        this.kotlinOptions {
+            jvmTarget = "1.8"
+            freeCompilerArgs = listOf("-opt-in=kotlinx.serialization.ExperimentalSerializationApi")
+        }
+    }
+
+    withType<Javadoc> {
+        val customArgs = projectDir.resolve("javadoc-silence.txt")
+        customArgs.writeText(
+            """-Xdoclint:none
+            """.trimIndent()
+        )
+        options.optionFiles?.add(customArgs)
+    }
+
+    withType<Test> {
```

Comment:
As far as I remember this is eager, @chao2zhang, configureEach is the solution.
To validate, run `gradlew --scan`, in the Performance section it should show 0 (max 1 = :help) task created during configuration.

Reply:
You are absolutely right. We could introduce https://github.com/autonomousapps/gradle-best-practices-plugin into not only this repo but also the main detekt repo.

This issue shouldn't impact the `sarif4k:0.2.0` though.

Reply:
I agree it doesn't directly affect this repo, but if you consider that this repo and Detekt is seen by a lot of people who use Gradle (e.g. Android devs) they might copy these patterns as "best practices", which is not good for any project.

I agree about checks to verify the builds, finally the community is trying to solve the "Big Gradle confusion", IntelliJ also added some static checks. I don't think a Gradle plugin is the best to detect these though, Detekt or other static analysis tools (probably PMD) might be better equipped. The plugin you mentioned has a good goal, but right now it's early stages: https://twitter.com/AutonomousApps/status/1571554472555581440

This is a way more comprehensive list of best practices to apply:
https://github.com/liutikas/gradle-best-practices

Reply:
Fixed in https://github.com/detekt/sarif4k/commit/c6a5c5ea08d8965ab3a00595fcdcf42d64363bf4 🙏